src/sql/workbench/api/node/mainThreadConnectionManagement.ts

```diff
@@ -70,6 +70,9 @@ export class MainThreadConnectionManagement implements MainThreadConnectionManag
 		}
 		let connectionProfile = await this._connectionDialogService.openDialogAndWait(this._connectionManagementService,
 			{ connectionType: connectionType, providers: providers }, initialConnectionProfile, undefined);
+		if (connectionProfile) {
+			connectionProfile.options.savePassword = connectionProfile.savePassword;
+		}
```

Comment:
Why do you have to pass this down through? It seems like a bug to not have these already be the same.

Reply:
Because right now there's no way to tell whether the "Remember password" was checked or not in the connection dialog from the API.